enterprise/cypher/physical-planning/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/SlotAllocation.scala

```diff
@@ -304,6 +304,16 @@ object SlotAllocation {
         }
         newPipeline
 
+      case _: ValueHashJoin =>
+        val newPipeline = lhsPipeline.breakPipelineAndClone()
```

Comment:
This could also be done by forwarding the RHS pipeline

Reply:
Not sure what you mean here, @henriknyman. Are you suggesting we change something?